sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala

```diff
@@ -137,11 +137,12 @@ object EvaluatePython {
       case c: Int => c
     }
 
-    case TimestampType | TimestampNTZType => (obj: Any) => nullSafeConvert(obj) {
-      case c: Long => c
-      // Py4J serializes values between MIN_INT and MAX_INT as Ints, not Longs
-      case c: Int => c.toLong
-    }
+    case TimestampType | TimestampNTZType | _: DayTimeIntervalType => (obj: Any) =>
```

Comment:
e.g.) PySpark reads internal data format of `DayTimeIntervalType` as `Long`, and convert it to Python instances at https://github.com/apache/spark/pull/34614/files#diff-bb02965370d515ec7add7967bc949c18c88ffc7346cceaebefb84907dfdce903R376 for both createDataFrame and Python UDF